tooling/Microsoft.VisualStudio.BlazorExtension/LiveReloadBuildWatcher.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.VisualStudio.BlazorExtension
+{
+    internal class LiveReloadBuildWatcher : IVsUpdateSolutionEvents2
+    {
+        const string BlazorProjectCapability = "Blazor";
+
+        private bool _isListeningForProjectBuilds = false;
+        private List<string> _signalFilePathsToNotify = new List<string>();
+
+        public int UpdateSolution_Begin(ref int pfCancelUpdate)
+        {
+            _signalFilePathsToNotify.Clear();
+            _isListeningForProjectBuilds = true;
+            return VSConstants.S_OK;
+        }
+
+        public int UpdateSolution_Done(int fSucceeded, int fModified, int fCancelCommand)
+        {
+            _isListeningForProjectBuilds = false;
+
+            foreach (var fullPath in _signalFilePathsToNotify)
+            {
+                File.WriteAllText(fullPath, string.Empty);
+                File.Delete(fullPath);
+            }
```

Comment:
You should probably add some error handling and/or retries here. If you have any sort of failures here like a long path, permissions issue, file locking, etc I think this will crash VS.